### ISSUE TITLE
`examples/sync-rebuild` doesn't work with `--chroot`

### ISSUE THREAD

**Issue by FichteFoll** [AladW/aurutils#1096]
I just updated to the latest version and wanted to test the new chroot mode of sync-rebuild but it errors.

```
~/dotfiles ▞ aur sync-rebuild vencord-desktop-git -c -d aur
==> Using [aur] repository
    vencord-desktop-git r116.f58ed48-1 is newer than r67.d1acb04-2
dependency vencord satisfied by aur/vencord-desktop-git
==> Retrieving package files
Already up to date.
/home/fichte/.local/state/aurutils/sync/ninja-fichte-2925105
Traceback (most recent call last):
  File "/home/fichte/bin/aur-sync-rebuild", line 333, in <module>
    main({i:1 for i in args.targets}, args.database, aurdest,
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fichte/bin/aur-sync-rebuild", line 264, in main
    repo_targets_ordered[name] = repo_targets[name]
                                 ~~~~~~~~~~~~^^^^^^
KeyError: 'vencord-desktop-git\tvencord-desktop-git'
```

**Comment by FichteFoll** [comment]
I also tried this with the `--user` parameter but that resulted in a different failure.
```
~/dotfiles ▚ sudo aur-sync-rebuild vencord-desktop-git -c -d aur -U fichte
[sudo] password for fichte:
==> Using [aur] repository
    vencord-desktop-git r116.f58ed48-1 is newer than r67.d1acb04-2
dependency vencord satisfied by aur/vencord-desktop-git
==> Retrieving package files
Already up to date.
Traceback (most recent call last):
  File "/home/fichte/bin/aur-sync-rebuild", line 333, in <module>
    main({i:1 for i in args.targets}, args.database, aurdest,
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fichte/bin/aur-sync-rebuild", line 271, in main
    rebuilds, failed = rebuild_packages(repo_targets_ordered, db_name, start_dir,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fichte/bin/aur-sync-rebuild", line 131, in rebuild_packages
    src_pkgver, _ = srcinfo_get_version(os.path.join(src_dir, '.SRCINFO'))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fichte/bin/aur-sync-rebuild", line 45, in srcinfo_get_version
    with open(srcinfo, 'r', encoding='utf-8') as file:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/root/.cache/aurutils/sync/vencord-desktop-git/.SRCINFO'
```

**Comment by AladW** [comment]
I can't replicate the first issue, the second is because `AURDEST` should be set explicitly. If you know of a way to use `pyxdg.BaseDirectory` with a different user, I'm happy to make the change

**Comment by FichteFoll** [comment]
> I can't replicate the first issue.

You may have to manually/explicitly build an older commit of the package first (e.g. the one referenced in my log). My best guess would be to pin the commit in the source url, e.g. `#commit=f58ed48`. If you tried that already, no clue.

> If you know of a way to use `pyxdg.BaseDirectory` with a different user, I'm happy to make the change.

It appears the current user is more or less [hard-coded at module import time](https://gitlab.freedesktop.org/xdg/pyxdg/-/blob/1d23e483ae869ee9532aca43b133cc43f63626a3/xdg/BaseDirectory.py#L30), so a working workaround would be to change `os.environ['HOME']` *before* importing the module. Pretty hacky, though …

```py
In [4]: os.environ['HOME'] = "/tmp"

In [5]: os.path.expanduser('~')
Out[5]: '/tmp'
```

[xdg-base-dirs](https://pypi.org/project/xdg-base-dirs/) evaluates the user's HOME path on each method call, so it would be easier to mess with. Figuring out a certain user's HOME path would still need to be done, though. Or requested as a feature for xdg-base-dirs.

> it looks like your version of `sync-rebuild` reads `aur sync --columns` output for some reason

Well, at least I didn't modify it.

```
~/bin Λ diff -u /usr/share/doc/aurutils/examples/sync-rebuild ~/bin/aur-sync-rebuild

~/bin > pacman -Qs aurutils
local/aurutils 16.5-1
    helper tools for the arch user repository

~/bin λ which aur-sync-rebuild
/home/fichte/bin/aur-sync-rebuild
```

**Comment by AladW** [comment]
> My best guess would be to pin the commit in the source url, e.g. #commit=f58ed48

I will test that later. For the rest, I came up with this:
```python
def xdg_cache_home(user=None):
    """Retrieve XDG_CACHE_HOME from the XDG Base Directory specification
    """
    if user is not None:
        user_home = os.path.expanduser("~" + user)
    else:
        user_home = os.path.expanduser("~")
    cache_home = os.path.join(user_home, '.cache')

    if 'XDG_CACHE_HOME' in os.environ:
        return os.getenv('XDG_CACHE_HOME')

    return cache_home
```
The only issue is that `os.environ` only works for the current user, so if `AURDEST` or `XDG_CACHE_HOME` is set in the `--user` environment, it would not be retrieved.

**Comment by FichteFoll** [comment]
- Second error was fixed.
- If I run the first command after the second, a stray `PKGBUILD.tmp` created by the root user remains in the AURDEST/package folder, which leads to:
```
~ λ aur sync-rebuild vencord-desktop-git -c -d aur
==> Using [aur] repository
    vencord-desktop-git r116.f58ed48-1 is newer than r67.d1acb04-2
dependency vencord satisfied by aur/vencord-desktop-git
==> Retrieving package files
Already up to date.
Traceback (most recent call last):
  File "/home/fichte/bin/aur-sync-rebuild", line 362, in <module>
    main({i:1 for i in args.targets}, args.database, aurdest,
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fichte/bin/aur-sync-rebuild", line 299, in main
    rebuilds, failed = rebuild_packages(repo_targets_ordered, db_name, start_dir,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fichte/bin/aur-sync-rebuild", line 162, in rebuild_packages
    shutil.copy(buildscript, buildscript + backup)
  File "/usr/lib/python3.11/shutil.py", line 419, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.11/shutil.py", line 258, in copyfile
    with open(dst, 'wb') as fdst:
         ^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: '/home/fichte/.cache/aurutils/sync/vencord-desktop-git/PKGBUILD.tmp'
```
after removing the file manually, the first command also works.
-- `--vcs` also works, though I suppose it would make more sense to call it `--pkgver` to be more in line with aur-sync?

**Comment by AladW** [comment]
Fixed in 17.2
